src/core/pointcloud/qgspointcloudattribute.h

```diff
@@ -96,11 +96,23 @@ class CORE_EXPORT QgsPointCloudAttributeCollection
     const QgsPointCloudAttribute *find( const QString &attributeName, int &offset ) const;
 
     //! Returns total size of record
-    int pointRecordSize() const { return mSize; }
+    std::size_t pointRecordSize() const { return mSize; }
```

Comment:
not a fan... we work with qt containers that all work on int indices (https://doc.qt.io/qt-5/qvector.html#size). not sure if we win anything here by using site_t?

Reply:
Qt 6 compatibility for one :P

Reply:
But the actual rationale here was to match the type of QgsPointCloudAttribute::size() -- if we mix it gets annoying to deal with things which are sensitive to the type, like the QCOMPARE macro (which breaks the build on travis for some reason and doesn't auto cast to int)